js/src/client.ts

```diff
@@ -711,6 +725,40 @@ export class Client {
       }
       return;
     }
+    const sizeLimitBytes =
+      this.serverInfo?.batch_ingest_config?.size_limit_bytes ??
+      DEFAULT_BATCH_SIZE_LIMIT_BYTES;
+    const batchChunks = {
+      post: [] as (typeof rawBatch)["post"],
+      patch: [] as (typeof rawBatch)["patch"],
+    };
+    let currentBatchSizeBytes = 0;
+    for (const k of ["post", "patch"]) {
+      const key = k as keyof typeof rawBatch;
```

Comment:
oh this syntax is weird to me - cool

Reply:
It's just shorthand so we don't have to cast everywhere